core/schema.py

```diff
@@ -508,8 +522,19 @@ def orderBy(cls, qs, args):
     def resolve_queryset(
             cls, connection, iterable, info, args, filtering_args, filterset_class
     ):
+        request = getattr(info, "context", None)
+
+        csrf_token = request.headers.get("X-CSRFToken")
+        requested_with = request.headers.get("X-Requested-With")
+        stored_token = cache.get(f"csrf_token_{request.user.id}")
+        if requested_with == WEBAPP_EXPECTED_REQUESTED_WITH:
```

Comment:
Here we check source of request, if web app - then csrf is checked @delcroip

Reply:
No, the goal it to enforce it for all but few exception, because with your approach a JS script doing a crosssite request forgery won't be stopped

Reply:
In that case, should we add check if 'if requested_with == 'Android': <skip check>'? Is this better?